sdk/trace/trace_test.go

```diff
@@ -217,13 +217,35 @@ func TestSetName(t *testing.T) {
 	}
 }
 
-func TestRecordingIsOn(t *testing.T) {
-	tp := NewTracerProvider()
-	_, span := tp.Tracer("Recording on").Start(context.Background(), "StartSpan")
-	defer span.End()
-	if span.IsRecording() == false {
-		t.Error("new span is not recording events")
-	}
+func TestSpanIsRecording(t *testing.T) {
+	t.Run("Defer ending the span", func(t *testing.T) {
+		for name, tc := range map[string]struct {
+			sampler Sampler
+			want    bool
+		}{
+			"Always sample, recording on": {sampler: AlwaysSample(), want: true},
+			"Never sample recording off":  {sampler: NeverSample(), want: false},
+		} {
+			tp := NewTracerProvider(WithSampler(tc.sampler))
+			_, span := tp.Tracer(name).Start(context.Background(), "StartSpan")
+			defer span.End()
+			got := span.IsRecording()
+			assert.Equal(t, got, tc.want, name)
+		}
+	})
+
+	t.Run("End span immediately", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("after Span end", func(t *testing.T) {
```

This will read in the test output as something like `TestSpanIsRecording/after_Span_end`, so it makes sense to try to make them read naturally.

Reply:
Oh that's a neat concept to improve readability.